jxl/src/simd/mod.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+use std::{
+    fmt::Debug,
+    ops::{Add, AddAssign, Mul, MulAssign, Sub, SubAssign},
+};
+
+#[cfg(target_arch = "x86_64")]
+mod x86_64;
+
+mod scalar;
+
+#[cfg(target_arch = "x86_64")]
+pub(crate) use x86_64::{avx::AvxDescriptor, avx512::Avx512Descriptor, simd_function};
+
+#[cfg(not(target_arch = "x86_64"))]
+pub(crate) use scalar::simd_function;
+
+#[cfg(all(test, target_arch = "x86_64"))]
+pub(crate) use x86_64::test_all_instruction_sets;
+
+#[cfg(all(test, not(target_arch = "x86_64")))]
+pub(crate) use scalar::test_all_instruction_sets;
+
+pub(crate) use scalar::ScalarDescriptor;
+
+pub trait SimdDescriptor: Sized + Copy + Debug + Send + Sync {
+    type F32Vec: F32SimdVec<Descriptor = Self>;
+
+    fn new() -> Option<Self>;
+}
+
+pub trait F32SimdVec:
+    Sized
+    + Copy
+    + Debug
+    + Send
+    + Sync
+    + Add<Self, Output = Self>
+    + Mul<Self, Output = Self>
+    + Sub<Self, Output = Self>
+    + AddAssign<Self>
+    + MulAssign<Self>
+    + SubAssign<Self>
+{
+    type Descriptor: SimdDescriptor;
+
+    const LEN: usize;
+
+    fn splat(d: Self::Descriptor, v: f32) -> Self;
```

Comment:
could you add a comment here? I looked up the dictionary definition of splat: "a single flat thin often ornamental member of a back of a chair" which does not seem to be related...